### PR TITLE
Fix #126: use real user id in batch jobs instead of proxy user id

### LINKED ISSUE #126: `load_result` not working in batch jobs when using OIDC user
originally reported here https://discuss.eodc.eu/t/caching-openeo-results/197/22

using `load_result` fails with something like
```
 openeo_driver.errors.JobNotFoundException: The batch job ‘4cfc3e73-4847-483f-a580-34d765e86f30’ does not exist.
```

after some digging, I found that the job lookup inside a batch job does not work because the batch job uses the proxyuser as user id while the job registry uses (OIDC based) user id

### PR DESCRIPTION
**Pull request by soxofaan**
see #126 

openEO processing in batch jobs needs real user id for `load_result` and UDPs to work properly.
Proxy user id is just necessary when submitting the job.

### CODE REVIEW

**Comment by soxofaan** [comment]
@bossie   do you think this is going to work?

File: openeogeotrellis/deploy/batch_job.py [hunk 1]
```diff
@@ -239,7 +239,9 @@ def main(argv: List[str]) -> None:
     metadata_file = job_dir / argv[5]
     api_version = argv[6] if len(argv) >= 7 else None
     dependencies = _deserialize_dependencies(argv[7]) if len(argv) >= 8 else {}
-    user_id = argv[8] if len(argv) >= 9 else None
+    # TODO: do we still need proxy user id inside the batch job itself?
+    proxy_user_id = argv[8] if len(argv) >= 9 else None
```
Comment (bossie) [inline openeogeotrellis/deploy/batch_job.py:243]: The proxy_user_id argument can be removed.

Reasoning:

The proxy user ID is only used to submit a Spark job (spark-submit --proxy-user ...) which only happens in two situations:
- explicit start job in the web API (POST '/jobs/<job_id>/results');
- asynchronously when SHub dumps batch process results in an S3 bucket and we can act on them.

So a batch job never submits batch jobs itself.

File: openeogeotrellis/deploy/submit_batch_job.sh [hunk 2]
```diff
@@ -134,4 +134,4 @@ spark-submit \
  --conf spark.yarn.tags=openeo \
  --jars "${extensions}","${backend_assembly}" \
  --name "${jobName}" \
- "${main_py_file}" "$(basename "${processGraphFile}")" "${outputDir}" "${outputFileName}" "${userLogFileName}" "${metadataFileName}" "${apiVersion}" "${dependencies}" "${proxyUser}"
+ "${main_py_file}" "$(basename "${processGraphFile}")" "${outputDir}" "${outputFileName}" "${userLogFileName}" "${metadataFileName}" "${apiVersion}" "${dependencies}" "${proxyUser}" "${userId}"
```
Comment (bossie) [inline openeogeotrellis/deploy/submit_batch_job.sh:137]: "${proxyUser}" argument can be removed.

File: openeogeotrellis/deploy/submit_batch_job_spark24.sh [hunk 3]
```diff
@@ -144,4 +144,4 @@ spark-submit \
  --conf spark.yarn.tags=openeo \
  --jars "${extensions}","${backend_assembly}" \
  --name "${jobName}" \
- "${main_py_file}" "$(basename "${processGraphFile}")" "${outputDir}" "${outputFileName}" "${userLogFileName}" "${metadataFileName}" "${apiVersion}" "${dependencies}" "${proxyUser}"
+ "${main_py_file}" "$(basename "${processGraphFile}")" "${outputDir}" "${outputFileName}" "${userLogFileName}" "${metadataFileName}" "${apiVersion}" "${dependencies}" "${proxyUser}" "${userId}"
```
Comment (bossie) [inline openeogeotrellis/deploy/submit_batch_job_spark24.sh:147]: "${proxyUser}" argument can be removed.

File: openeogeotrellis/deploy/submit_batch_job_spark3.sh [hunk 4]
```diff
@@ -165,4 +165,4 @@ spark-submit \
  --conf spark.yarn.tags=openeo \
  --jars "${extensions}","${backend_assembly}" \
  --name "${jobName}" \
- "${main_py_file}" "$(basename "${processGraphFile}")" "${outputDir}" "${outputFileName}" "${userLogFileName}" "${metadataFileName}" "${apiVersion}" "${dependencies}" "${proxyUser}"
+ "${main_py_file}" "$(basename "${processGraphFile}")" "${outputDir}" "${outputFileName}" "${userLogFileName}" "${metadataFileName}" "${apiVersion}" "${dependencies}" "${proxyUser}" "${userId}"
```
Comment (bossie) [inline openeogeotrellis/deploy/submit_batch_job_spark3.sh:168]: "${proxyUser}" argument can be removed.

**Review by bossie — CHANGES_REQUESTED** [review]
I would remove the "${proxyUser}" argument to batch_job.py as it is not needed.

**Comment by soxofaan** [comment]
yes indeed, I didn't see `proxyUser` being used in batch_job.py, but I wanted to be sure it was fine to remove/replace it

**Pull request merged**